pkg/test/runner/config.go

```diff
@@ -54,7 +54,9 @@ type EvalTestCaseConfig struct {
 type TestCaseConfig struct {
 	// ExitCode is the expected exit code from the kpt commands. Default: 0
 	ExitCode int `json:"exitCode,omitempty" yaml:"exitCode,omitempty"`
-
+	// StdErr is the expected standard error output and should be checked
+	// when a nonzero exit code is expected. Default: ""
+	StdErr string `json:"stdErr,omitempty" yaml:"stdErr,omitempty"`
```

Comment:
I think we should check stdout as well. And if content in stdout/stderr is pretty long or multiple lines, a string field may not very good.

Reply:
what type for stdout/stderr would be good, if not string?

Reply:
Maybe put them in `cli_output.txt` or `command_output.txt` in `.expected`? How do you think?

Reply:
would we want two files (e.g. cli_stderr.txt, cli_stdout.txt), or combined into one?

Reply:
I think 2 files are clearer. We should go this way. And also please remember to add them to batch update function.

Reply:
Two things here:

- YAML support [multiline ](https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines) so specifying big text should be ok. Having separate files increases the burden on the test author, and reviewer. Having expected output in a single place as much as possible helps. So lets avoid multiple files here.
- Another thing I was thinking was to use using this as `substring` instead of exact match (haven't looked at the entire PR yet). In the future, we can treating this as Go template (that supports `PkgPath` as a variable) to allow test authors to include pkg-path also in the message.

Reply:
Using a multiple line string in `config.yaml` has some disadvantages:
1. We cannot control the format when we automatically update the expected results. It maybe encoded to a single line string. (Not sure can we control the format)
2. If the output is very long, most parts of `config.yaml` will be the output string and it's also hard to read. Although not sure will we have a very long output in the test.

Reply:
Output doesn't seem too long (and can be shorter if we use substrings). I will add checking stdout here and see what happens

Reply:
1. I updated the PR to also check for stdout. Aligning with @droot's suggestion, I haven't moved it out to a new file for now but can do so if that is more desirable. 

2. @Shell32-Natsu If we move stdout/stderr into separate files I can add them to the batch update function.

However, with them in config.yaml I think it will be nontrivial to update. 

The batch update function doesn't update config.yaml currently - e.g. if it has the wrong expected exit code this needs to be changed manually AFAICT (though please correct me if I'm wrong). 

Updating config.yaml while preserving comments and not including defaults etc might take some time to figure out (and I think should be a separate issue/PR).

Reply:
1. I've updated to check the output of stdout as well. Aligning with @droot's suggestion, I've left it in config.yaml for now. 

2. @Shell32-Natsu is there an easy way to update config.yaml? Currently `updateExpected` doesn't seem to change it - e.g. if the expected exit code is wrong it has to be changed manually AFAICT (though please correct me if I'm wrong). I can write the output of yaml.Marshal(r.testCaseConfig) but that produces some unnecessary and unwanted output. Unless I'm missing an easy way to do the update, I suggest we file a separate issue for it because I don't think it is as high priority as some other tasks right now.

Reply:
Yes, if we plan to auto update like we do for `diff` and `results`, specifying them in separate files will be preferred.
I was thinking of using substring of the expected output (so expected them to be short) and avoiding regular updates, so wasn't thinking about batch auto update will be needed, but I could be wrong. If go down the route of supporting matching the exact output, then we will have to support template (pkgPath substitution might be needed) and keeping them in separate files.

WDYT ?

Reply:
Yes, I am already using a substring of the expected output (checking if actual output contains the expected output). 

https://github.com/GoogleContainerTools/kpt/blob/c91da1d1cabef6cbe3809bca27f9c41b1a0813e6/pkg/test/runner/runner.go#L312

The actual output is longer (has multiple lines and includes the package name). If we check for exact match, separate files may be preferable as I was having some trouble with the multilines string. Personally I like this approach; it is simpler and easier to read, and as you've said, will avoid regular updates.

Reply:
I also think `substring` match is good enough for our use-case here. Lets go with this and if it doesn't work for us in the future, we can always address it by supporting separate files.